### PR TITLE
MNT: use mod to wrap index

### PR DESCRIPTION
**Pull request by tacaswell**
Marginally safer than an if statement.

### CODE REVIEW

**Comment by tacaswell** [comment]
Probably a bit of both.

The sketch I wrote up in the issues tried to be fixed time windows, this is always the last N points.  Not clear which one is better....which probable means we want both!

File: pcdsdevices/signal.py [hunk 1]
```diff
@@ -174,7 +174,6 @@ def _update_avg(self, *args, value, **kwargs):
         with self._lock:
             self.values[self.index] = value
             self.index += 1
```
Comment (ZLLentz) [inline pcdsdevices/signal.py]: We're incrementing the index twice now, so we should delete this line.
```suggestion
```

File: pcdsdevices/signal.py [hunk 2]
```diff
@@ -174,7 +174,6 @@ def _update_avg(self, *args, value, **kwargs):
...
-            if self.index == len(self.values):
-                self.index = 0
+            self.index = (self.index + 1) % len(self.values)
```
Comment (ZLLentz) [inline pcdsdevices/signal.py]: Trailing whitespace:
```suggestion
            self.index = (self.index + 1) % len(self.values)
```

**Comment by teddyrendahl** [comment]
@tacaswell You mind accepting @ZLLentz suggestions? Both will be required to get this to pass on our CI

**Comment by tacaswell** [comment]
fair enough.

That is what I get for trying to do the PR in the github editor

**Pull request merged**